Tests/LDSTTest.lean

```diff
@@ -0,0 +1,159 @@
+/-
+Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author(s): Yan Peng
+-/
+
+import Arm.Exec
+
+section LDSTTest
+
+open Std.BitVec
+
+-- This file contains unit tests for LDST instructions
+-- The cosimulation tests do not cover instructions related to memory access
+-- TODO: use macros to simplify the tests
+
+def set_init_state (find? : Store (BitVec 64) (Option (BitVec 32))) : ArmState :=
+  let s := { gpr := (fun (_ : BitVec 5) => 0#64),
+             sfp := (fun (_ : BitVec 5) => 0#128),
+             pc := 0#64,
+             pstate := (fun (_ : PFlag) => 0#1),
+             mem := (fun (_ : BitVec 64) => 0#8),
+             program := find?,
+             error := StateError.None}
+  s
+
+----------------------------------------------------------------------
+-- test ldr, 32-bit gpr, unsigned offset
+def ldr_gpr_unsigned_offset : program :=
+  def_program
+    #[ (0x0#64, 0xb9400fe0#32) ]  -- ldr w0, [sp, #12]
+
+def ldr_gpr_unsigned_offset_state : ArmState :=
+  let s := set_init_state ldr_gpr_unsigned_offset.find?
+  -- write 20 in 4 bytes to address 12
+  let s := write_mem_bytes 4 12#64 20#32 s
+  s
+
+def ldr_gpr_unsigned_offset_final_state : ArmState := run 1 ldr_gpr_unsigned_offset_state
+
+example : (read_store 0#5 ldr_gpr_unsigned_offset_final_state.gpr) = 20#64 := by native_decide
```

Comment:
Any reason why you didn't use `read_gpr`? I'm hoping that the API to access the Arm state is via these kinds of accessors or updaters, or via the more generic `r`/`w` functions.

Reply:
Fixed.